src/DcmTrajectoryGenerator.cpp

```diff
@@ -0,0 +1,652 @@
+{
+
+  m_trajectoryDomain = std::make_pair(phaseShift.front(), phaseShift.back());
+  m_phaseShift = phaseShift;
+  m_orderedSteps = orderedSteps;
+
+  // reset the trajectory
+  m_trajectory.clear();
+  
+  // add the first stance foot at the beginning of the orderedStep vector
+  m_orderedSteps.insert(m_orderedSteps.begin(), firstStanceFoot);
+  
+
+  double singleSupportStartTime;
+  double singleSupportEndTime;
+  double singleSupportDuration;
+  double doubleSupportEndTime;
+  double singleSupportT0;
+
+  // evaluate times for the last step
+  getLastStepsTiming(singleSupportStartTime, singleSupportEndTime,
+		     singleSupportDuration, doubleSupportEndTime,
+		     singleSupportT0);
+
+
+  iDynTree::Vector2 lastZmp = m_orderedSteps.back()->position;
+  m_orderedSteps.pop_back();
+
+  // evaluate the position of the Center of mass at the end of the trajectory 
+  iDynTree::Vector2 comPosition;
+  iDynTree::toEigen(comPosition)  = (iDynTree::toEigen(lastZmp) + iDynTree::toEigen(m_orderedSteps.back()->position)) / 2;
+
+  lastZmp = m_orderedSteps.back()->position;
+  m_orderedSteps.pop_back();
+
+  // evaluate the last step
+  if(!addLastStep(singleSupportStartTime, singleSupportEndTime,
+		  singleSupportDuration, doubleSupportEndTime,
+		  singleSupportT0,
+		  comPosition, lastZmp)){
+    std::cerr << "[DCM TRAJECTORY GENERATOR] Error when the DCM trajectory of the last step is generated." << std::endl;
+    return false;
+  }
+  
+  while (m_orderedSteps.size() > 0){
+    
+    getStepsTiming(singleSupportStartTime, singleSupportEndTime,
+  		   singleSupportDuration, singleSupportT0);
+
+    
+    lastZmp = m_orderedSteps.back()->position;
+    m_orderedSteps.pop_back();
+    
+    if(!addNewStep(singleSupportStartTime, singleSupportEndTime,
+  		   singleSupportDuration, singleSupportT0, 
+  		   lastZmp)){
+      std::cerr << "[DCM TRAJECTORY GENERATOR] Error when the DCM trajectory of a new step is generated." << std::endl;
+      return false;
+    }
+  }
+
+  // evaluate the first Double support phase
+  double doubleSupportStartTime;
+  getFirstDoubleSupportTiming(doubleSupportStartTime);
+
+  if(!addFirstDoubleSupportPhase(doubleSupportStartTime,
+  				 initPosition,
+  				 initVelocity)){
+    std::cerr << "[DCM TRAJECTORY GENERATOR] Error when the DCM trajectory of the first DS phase is generated." << std::endl;
+    return false;
+  }
+
+  // evaluate the DCM trajectory
+  if(!evaluateDcmTrajectory()){
+    std::cerr << "[DCM TRAJECTORY GENERATOR] Error when the whole DCM trajectory is evaluated." << std::endl;
+    return false;
+  }
+
+  return true;
+}
+
+bool DcmTrajectoryGenerator::evaluateDcmTrajectory()
+{
+  size_t timeVectorLength = std::get<1>(m_trajectoryDomain) - std::get<0>(m_trajectoryDomain);
+  std::vector<size_t> timeVector(timeVectorLength);
+  
+  // populate the time vector
+  std::iota(std::begin(timeVector), std::end(timeVector), std::get<0>(m_trajectoryDomain));
```

Comment:
This is not clear to me. ``std::get<0>(m_trajectoryDomain)`` is a ``double`` and ``iota`` applies the operator ``++`` on it. Usually ``++`` means "increase to the next value" thus its extension to double may not be very clear. In addition I don't understand why the ``timeVector`` should be filled with steps of 1.

Reply:
`m_trajectoryDomain` is a `std::pair<size_t, size_t>` ([here](https://github.com/GiulioRomualdi/unicycle-footstep-planner/blob/DCM_devel/include/DcmTrajectoryGenerator.h#L237) you can find the declaration) and its first value is the the first value of the `phaseShift` vector while the second value is the last value of the `phaseShift` vector (see [here](https://github.com/GiulioRomualdi/unicycle-footstep-planner/blob/DCM_devel/src/DcmTrajectoryGenerator.cpp#L495)). The `phaseShift` is a vector containing all the indices when a phase shift occurs. 
The `timeVector` object contains all the indices at which time the DCM trajectory [will be evaluate](https://github.com/GiulioRomualdi/unicycle-footstep-planner/blob/DCM_devel/src/DcmTrajectoryGenerator.cpp#L595). It is important to notice that it is only used in the [`for()` loop](https://github.com/GiulioRomualdi/unicycle-footstep-planner/blob/DCM_devel/src/DcmTrajectoryGenerator.cpp#L594), I think that it is better write a standard `for` loop instead instantiate the `timeVector` object.
```c++
size_t startTime = std::get<0>(m_trajectoryDomain);
size_t endTime = std::get<1>(m_trajectoryDomain);
for (size_t t = startTime ;  t < endTime ; t++){
   ...
}
```

Reply:
Ok sorry I did not see it! I was mislead by the [line 35](https://github.com/GiulioRomualdi/unicycle-footstep-planner/blob/DCM_devel/include/DcmTrajectoryGenerator.h#L35). I agree on the new version of the for, but I think that the term ``time`` and ``t`` are misleading since they recall continuous  quantities.